libs/cardano-ledger-test/src/Test/Cardano/Ledger/STS.hs

```diff
@@ -157,17 +183,18 @@ conwayProof = Conway Standard
 -- PParams ----------------------------------------------------------------
 
 pParamsT ::
+  Reflect era =>
   RootTarget
-    (ConwayEra StandardCrypto)
-    (PParamsF (ConwayEra StandardCrypto))
-    (PParamsF (ConwayEra StandardCrypto))
-pParamsT = Lensed (pparams conwayProof) id
+    era
+    (PParamsF era)
+    (PParamsF era)
+pParamsT = Lensed (pparams reify) id
 
-pParamsPs :: Subst (ConwayEra StandardCrypto) -> [Pred (ConwayEra StandardCrypto)]
-pParamsPs s = applySubst s $ pParamsPreds conwayProof
+pParamsPs :: Reflect era => Subst era -> [Pred era]
+pParamsPs s = applySubst s $ pParamsPreds reify
 
-pParamsGen :: Subst (ConwayEra StandardCrypto) -> Gen (PParamsF (ConwayEra StandardCrypto))
-pParamsGen s = genFromConstraints conwayProof standardOrderInfo (pParamsPs s) pParamsT
+pParamsGen :: Reflect era => Subst era -> Gen (PParamsF era)
+pParamsGen s = genFromConstraints reify standardOrderInfo (pParamsPs s) pParamsT
 
 -- TODO: this is currently broken because it fails to find `maxTxSize`. This can be fixed
 -- by building a proper target for PParams but I don't yet know how to do that.
```

Comment:
Can't we use the 
 Proj :: Lens' b t -> Rep era t -> Term era b -> Term era t
on var      currPParams :: Era era => Proof era -> Term era (PParamsF era)
and the var
maxTxSize :: Era era => Proof era -> Term era Natural
to do something?

Reply:
I think we can do something like that. I've put a note in the file because it isn't blocking us from merging now.